### PR TITLE
Update free-courses-hi.md

### PR DESCRIPTION
**Pull request by AnkitaSikdar005**
## What does this PR do?
Add resource(s) 

## For resources
Added C language lectures in Hindi.

### Why is this valuable (or not)?
Yes, it is valuable.
### How do we know it's really free?
It is available on YouTube.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by AnkitaSikdar005** [comment]
@eshellman Please merge my pull request.

File: courses/free-courses-hi.md [hunk 1]
```diff
@@ -149,6 +149,7 @@
 ### <a id="c"></a>C
 
 * [All C Concepts \| Hindi](https://www.youtube.com/playlist?list=PL7ersPsTyYt1d8g5qaxbE6sjWDzs4D_1v) - Saurabh Shukla
+* [C Language for GATE](https://www.youtube.com/playlist?list=PLC36xJgs4dxG-IqARhc23jYTDMYt7yvZP) - GATE cse lectures by Amit Khurana
```
Comment (sohansai) [inline courses/free-courses-hi.md:152]:
```suggestion
* [C Language for GATE — Complete Playlist](https://www.youtube.com/playlist?list=PLC36xJgs4dxG-IqARhc23jYTDMYt7yvZP) - Gate CSE lectures by Amit Khurana
```

**Comment by AnkitaSikdar005** [comment]
@eshellman Please merge this pull request.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**